### PR TITLE
Linux build support

### PR DESCRIPTION
**Pull request by erorcun**

### CODE REVIEW

File: src/audio/oal/channel.cpp [hunk 1]
```diff
@@ -22,7 +22,7 @@ void CChannel::SetDefault()
 	Mix = 0.0f;
 		
 	Position[0] = 0.0f; Position[1] = 0.0f; Position[2] = 0.0f;
-	Distances[0] = 0.0f; Distances[1] = FLT_MAX;
+	Distances[0] = 0.0f; Distances[1] = 3.402823e+38; //FLT_MAX;
```
Comment (ShFil119) [inline src/audio/oal/channel.cpp:25]: FLT_MAX should be available in `<stdint.h>`.

Comment (ShFil119) [reply src/audio/oal/channel.cpp:25]: Sry `<float.h>`

File: src/control/Pickups.cpp [hunk 2]
```diff
@@ -988,7 +988,7 @@ INITSAVEBUF
 		aPickUps[i] = ReadSaveBuf<CPickup>(buf);
 
 		if (aPickUps[i].m_eType != PICKUP_NONE && aPickUps[i].m_pObject != nil)
-			aPickUps[i].m_pObject = CPools::GetObjectPool()->GetSlot((int32)aPickUps[i].m_pObject - 1);
+			aPickUps[i].m_pObject = CPools::GetObjectPool()->GetSlot((intptr_t)aPickUps[i].m_pObject - 1);
```
Comment (aap) [inline src/control/Pickups.cpp:991]: we have uintptr for these cases actually

File: src/control/Script.cpp [hunk 3]
```diff
@@ -507,14 +507,14 @@ void CRunningScript::Init()
 int open_script()
 {
 	static int scriptToLoad = 0;
-
+#ifdef _WIN32
 	if (GetAsyncKeyState('G') & 0x8000)
```
Comment (aap) [inline src/control/Script.cpp:511]: maybe fix this to use CPad or whatever finally. This code is incredibly old when I didn't have CPad yet.

Comment (erorcun) [reply src/control/Script.cpp:512]: can't be done, CGame::Initialise() blocks `keypressCB`

File: src/core/CdStreamPosix.cpp [hunk 4]
```diff
@@ -0,0 +1,516 @@
...
+CdStreamSync(int32 channel)
+{
+	ASSERT( channel < gNumChannels );
+	CdReadInfo *pChannel = &gpReadInfo[channel];
+	ASSERT( pChannel != nil );
+
+	if (flushStream[channel]) {
+#ifdef ONE_THREAD_PER_CHANNEL
+		pChannel->nSectorsToRead = 0;
+		pthread_kill(gpReadInfo[channel]->pChannelThread, SIGINT);
```
Comment (blingu) [inline src/core/CdStreamPosix.cpp:315]:
```suggestion
		pthread_kill(gpReadInfo[channel].pChannelThread, SIGINT);
```

**Pull request merged**